lmcache/v1/storage_backend/connector/mooncakestore_connector.py

```
@@ -152,33 +166,232 @@ def __init__(
+                    logger.warning(
+                        f"batch_get_into failed for key {keys[i]} (code={n_read})"
+                    )
+                    memory_objs[i].ref_count_down()
+                    continue
+
+                try:
+                    results[i] = self.reshape_partial_chunk(
+                        memory_objs[i],
+                        n_read,
+                    )
+                except Exception as exc:
+                    logger.error(f"Reshape failed for key {keys[i]}: {exc}")
+                    memory_objs[i].ref_count_down()
+
+            return results
+
+        except Exception as exc:
+            logger.error(f"batch_get_into threw exception: {str(exc)}")
+            # Release any buffers we successfully allocated
+            for i in valid_idx:
+                memory_objs[i].ref_count_down()
+            return [None] * len(keys)
+
+    async def _batch_get_buffer(
+        self, keys: List[CacheEngineKey]
+    ) -> List[Optional[MemoryObj]]:
+        """
+        Batch get using batch_get_buffer when metadata is stored remotely.
+        This is used when save_chunk_meta=True (metadata stored with data).
+        """
+        key_strs = [key.to_string() for key in keys]
+
+        try:
+            buffers = await asyncio.to_thread(self.store.batch_get_buffer, key_strs)
         except Exception as e:
-            logger.error(f"Failed to get key {key_str}. {e}")
+            logger.error(f"batch_get_buffer failed: {str(e)}")
+            return [None] * len(keys)
+
+        results = []
+        for i, buffer in enumerate(buffers):
+            if buffer is None:
+                logger.warning(f"Buffer {i} is None for key {key_strs[i]}")
+                results.append(None)
+                continue
+            try:
+                memory_obj = self._process_buffer_with_metadata(buffer)
+                results.append(memory_obj)
+            except Exception as e:
+                logger.error(
```

This is another place for silent error handling.